File: easy/1342.py
class Solution:
    def numberOfSteps (self, num: int) -> int:
        num_steps = 0
        
        while num != 0:
            if num % 2 == 0:
                num = num/2
                num_steps += 1
            elif num % 2 != 0:
                num -= 1
                num_steps += 1
        
        return num_steps

File: 1342.java
class Solution {
    public int numberOfSteps (int num) {
        int numSteps = 0;
        while (num !=0) {
            int switchNum = (num%2);
            switch (switchNum) {
                case 0:
                    num = (num/2);
                    numSteps += 1;
                    break;
                case 1:
                    num -= 1;
                    numSteps += 1;
                    break;
            }
        }
        return numSteps;
    }
}
